tools/createTestReport/create_test_report_in_rst.py

```diff
@@ -0,0 +1,169 @@
+"""Project specific Markdown converter functions.
+
+    Author: Andreas Merkle (andreas.merkle@newtec.de)
+"""
+
+# pyTRLCConverter - A tool to convert TRLC files to specific formats.
+# Copyright (c) 2024 - 2025 NewTec GmbH
+#
+# This file is part of pyTRLCConverter program.
+#
+# The pyTRLCConverter program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+#
+# The pyTRLCConverter program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with pyTRLCConverter.
+# If not, see <https://www.gnu.org/licenses/>.
+
+# Imports **********************************************************************
+from pyTRLCConverter.ret import Ret
+
+from pyTRLCConverter.rst_converter import RstConverter
+from pyTRLCConverter.trlc_helper import Record_Object
+
+# Variables ********************************************************************
+
+# Classes **********************************************************************
+
+
+class CustomRstConverter(RstConverter):
+    """Custom Project specific reStructuredText converter responsible for converting the
+        SW test case results into reStructuredText format.
+    """
+
+    def __init__(self, args: any) -> None:
+        """Initializes the converter.
+        """
+        super().__init__(args)
+
+        self._test_case_results = []
+
+    @staticmethod
+    def get_description() -> str:
+        """ Return converter description.
+
+         Returns:
+            str: Converter description
+        """
+        return "Convert SW test case results to reStructuredText format."
+
+    def leave_file(self, file_name: str) -> Ret:
+        """
+        Leave a file.
+
+        Args:
+            file_name (str): File name
+
+        Returns:
+            Ret: Status
+        """
+
+        # Multiple document mode?
+        if self._args.single_document is False:
+            assert self._fd is not None
+            self._print_test_case_results()
+
+        return super().leave_file(file_name)
+
+    # pylint: disable=unused-argument
+    def convert_record_object(self, record: Record_Object, level: int) -> Ret:
```

Comment:
This is the "old-style" sub-classing method.

New code should not overwrite this method, but install handlers for supported types.

Reply:
ok